sky/cli.py

```diff
@@ -1041,7 +1045,7 @@ def _make_task_or_dag_from_entrypoint_with_overrides(
     if is_yaml:
         assert entrypoint is not None
         usage_lib.messages.usage.update_user_task_yaml(entrypoint)
-        dag = dag_utils.load_chain_dag_from_yaml(entrypoint)
+        dag = dag_utils.load_chain_dag_from_yaml(entrypoint, env_overrides=env)
```

Comment:
Our original behavior is to ignore all the overrides, including `env`. Otherwise, the semantics will be a bit weird, i.e. overriding envs in all the tasks or just overriding the first task. Maybe we don't need this `env_overrides` option should not take effect for a dag with more than 1 task?

Reply:
Sounds good. Done.